core/src/tx.rs

```diff
@@ -95,7 +94,7 @@ impl AcceptedTransaction {
             }));
         }
 
-        if *iroha_genesis::GENESIS_ACCOUNT_ID == *tx.authority() {
+        if *iroha_genesis::GENESIS_DOMAIN_ID == *tx.authority().domain_id() {
             return Err(AcceptTransactionFail::UnexpectedGenesisAccountSignature);
```

Comment:
Not genesis account but genesis domain of the authority now causes transaction rejection